### ISSUE TITLE
Dynamic playlist loading

### ISSUE THREAD

**Issue by Protonosgit** [KRTirtho/spotube#859]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Pain

Large playlist sometimes take up to 2 minutes to load

### Suggested solution

Dynamically load the first 10-20 titles into the listview and then load more as the user scrolls down.
Or load them in chunks of the previously mentioned size so the user can start listening without having to wait for the whole playlist to load.

### Useful resources

You could check the last known size of the playlist and display empty loading placeholders if the user scrolls too quickly.

### Additional information

Alternatively just add an option to cache playlists and update them in the background or using a refresh button(android pull to refresh)

**Comment by KRTirtho** [comment]
Implemented in 28a5d6bb3820ab0bd4007664f73d685f6e1d2c90

**Comment by RokeJulianLockhart** [comment]
What https://github.com/KRTirtho/spotube/issues/859#issue-1977440804 describes isn't what I'd consider pagination - it's what I would consider dynamic loading. Pagination generally necessitates manual interaction to change pages. Can someone explain which type of loading was implemented?

**Comment by KRTirtho** [comment]
Yes, you can tell that, but is it really practical for a music app to load each page (only 20 track per page) manually?
To be exact, the current implementation is called Infinite scroll loading/pagination

**Comment by RokeJulianLockhart** [comment]
> https://github.com/KRTirtho/spotube/issues/859#issuecomment-1828973760

Thanks, @KRTirtho. That's good to hear. I hoped it would be infinite scrolling.
